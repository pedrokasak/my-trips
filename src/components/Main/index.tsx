import * as S from './styles'



const Main = () => <main>
    <S.Wrapper>
        <S.Logo src='/img/logo.svg' alt='Imagem de um átomo e react avançado escrito ao lado' />
        <S.Title>React Avançado</S.Title>
        <S.Description>Typescript, ReactJs, NextJs e Styled Components</S.Description>
        <S.Illustration  src='/img/hero-illustration.svg' alt='Imagem de um computador comuma pessoa sentada em uma cadeira utilizando o computador'/>
    </S.Wrapper>
</main>

export default Main