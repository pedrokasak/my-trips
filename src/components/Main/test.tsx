import { render, screen } from '@testing-library/react'
import Main from '.'


describe('<main />', () => {
    it('should render the heading', () => {
        render(<Main />)

        expect(screen.getByRole('heading', { name: /react avanÃ§ado/i })).toBeInTheDocument()
    })

    it('should render the heading', () => {
        const { container } = render(<Main />)

        expect(container.firstChild).toMatchSnapshot()
    })

    // it('should render colors correctly', () => {
    //     const { container } = render(<Main />)
    //     expect(container.firstChild).toHaveStyle('background-color: rgb(6, 9, 43)')

    // })
})
