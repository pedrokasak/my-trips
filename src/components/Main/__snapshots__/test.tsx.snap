// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<main /> should render the heading 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100vw;
  height: 100vh;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<main>
  <main
    class="c0"
  >
    <img
      alt="Imagem de um átomo e react avançado escrito ao lado"
      class="c1"
      src="/img/logo.svg"
    />
    <h1
      class="c2"
    >
      React Avançado
    </h1>
    <h2
      class="c3"
    >
      Typescript, ReactJs, NextJs e Styled Components
    </h2>
    <img
      alt="Imagem de um computador comuma pessoa sentada em uma cadeira utilizando o computador"
      class="c4"
      src="/img/hero-illustration.svg"
    />
  </main>
</main>
`;
